source/fundamentals/monitoring.txt

```diff
@@ -0,0 +1,409 @@
+==========
+Monitoring
+==========
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+.. What is on this page?
+
+This guide explains how you can set up and configure **monitoring** in the
+MongoDB Java driver.
+
+.. What do any new terms mean?
+
+Monitoring is the process of getting information about the activities a running 
+program performs for use in an application or an application performance
+management library. 
+
+Monitoring the MongoDB Java driver lets you understand the
+driver's resource usage and performance, and can help you make informed
+decisions when designing and debugging your application.    
+
+.. What can you expect to see on this page?
+
+In this guide you will learn how to perform these tasks:
+
+- :ref:`Monitor different types of events in the MongoDB Java Driver <monitoring-monitor-events>`
+- :ref:`Monitor connection pool events with Java Management Extensions (JMX) and JConsole <monitoring-jmx>`
+
+This guide shows how to use information about the activity of the driver in code.
+If you would like to learn how to record events the driver performs,
+consider reading our :doc:`guide on logging </fundamentals/logging>`.
+
+.. _monitoring-monitor-events:
+
+Monitor Events
+--------------
+
+To monitor an **event**, you must register a **listener** on your ``MongoClient``
+instance. 
+
+An event is any action that happens in a running program. The driver includes functionality
+for listening to a subset of the events that occur when the driver is running.
+
+A listener is a class that performs some action when certain events occur. All
+listener classes extend the ``EventListener`` interface. A listener's
+API defines the events it can respond to.
+
+Each method of a listener class represents a response to a certain event. Each
+method receives one argument: an object representing the event the method
+responds to. All event objects are a subclass of the ``EventObject`` class.  
+
+The MongoDB Java driver organizes the events it defines into three categories:
+
+- Command Events
+- Server Discovery and Monitoring Events
+- Connection Pool Events
+
+The following sections show how to monitor each event category.
+
+For a full list of the events you can monitor,
+:java-docs:`see the event package of the MongoDB Java Driver <apidocs/mongodb-driver-core/com/mongodb/event/package-summary.html>`.
+
+For more information on the classes and interfaces mentioned in this section,
+see the following API documentation:
+
+- `EventListener <https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/util/EventListener.html>`__
+- `EventObject <https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/util/EventObject.html>`__
+
+Command Events
```

Comment:
Add a note indicating that the driver does not publish events for commands called internally by the driver, e.g. for monitoring the cluster or for handshake on connection establishment, which includes `hello` and auth-related commands.

Reply:
Thanks for the suggestion. Here is the note: https://github.com/mongodb/docs-java/pull/88/commits/7bd930288fa4d60fbd8e9520f5c414b5602e658e

Let me know if you would like this note to go into more detail. I tried digging into the driver code to get an understanding of the specific types of commands the driver would not produce events for, but had trouble coming up with anything useful to readers. Apologies for not being able to fill out this content on my own, and please let me know if this is insufficient.

Reply:
This looks just fine.  Thanks.